modules/secure-logging/slog.c

```diff
@@ -420,15 +424,34 @@ gchar *convertToBase64(unsigned char *input, gsize len)
  */
 void cmac(unsigned char *key, const void *input, gsize length, unsigned char *out, gsize *outlen)
 {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+  EVP_MAC *mac = EVP_MAC_fetch(NULL, "CMAC", NULL);
+  OSSL_PARAM params[] =
+  {
+    OSSL_PARAM_utf8_string("cipher", "aes-256-cbc", 0),
+    OSSL_PARAM_END,
+  };
+
+  EVP_MAC_CTX *ctx = EVP_MAC_CTX_new(mac);
+
+  EVP_MAC_init(ctx, key, KEY_LENGTH, params);
+  EVP_MAC_update(ctx, input, length);
+  size_t out_len;
+  EVP_MAC_final(ctx, out, &out_len, 99999);
```

Comment:
This was probably an experiment, please fix the 4. parameter.

Reply:
Thank you, I completely missed it when I started working again on this branch